devito/ir/iet/visitors.py

```diff
@@ -542,13 +542,13 @@ def visit_Node(self, o, ret=None, parents=None, in_parent=False):
 
 class FindSymbols(Visitor):
 
-    def quick_str(o):
+    def quick_key(o):
         """
         SymPy's __str__ is horribly slow so we stay away from it for as much
         as we can. A devito.Indexed has its own overridden __str__, which
         relies on memoization, which is acceptable.
         """
-        return str(o) if o.is_Indexed else o.name
+        return (str(o) if o.is_Indexed else o.name, type(o))
```

Comment:
Why do you need the type now? In case two objects have same name?

Reply:
yep, this is a bug fix, probably explained in the relative commit message